rust/toolchain.bzl

```diff
@@ -301,3 +301,170 @@ See @rules_rust//rust:repositories.bzl for examples of defining the @rust_cpuX r
 with the actual binaries and libraries.
 """,
 )
+
+def _cargo_tool_impl(ctx):
```

Comment:
What do you think about a single rule that gets the name of the tool from an attribute, and then there would be a `if ctx.attr.tool` == "rustfmt":` conditional in the impl? I prefer that to duplicated logic in multiple files, even though they are so close to each other.

Reply:
Updated. Sad to say I'd forgotten about that aspect of this PR. The code I wrote this for made more sense to make the separate rules but I really like the introduction of the `current_*` targets since that's ultimately what seems to be in demand here. But the rule is now one rule which takes an additional argument.